drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,489 @@
+	if (temp < 0) {
+		SYS_LOG_DBG("Default integration time not set for ADC");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_WTIME_REG, APDS9960_DEFAULT_WTIME);
+	if (temp < 0) {
+		SYS_LOG_DBG("Default wait time not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PPULSE_REG, APDS9960_DEFAULT_PROX_PPULSE);
+	if (temp < 0) {
+		SYS_LOG_DBG("Default proximity ppulse not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_UR_REG, APDS9960_DEFAULT_POFFSET_UR);
+	if (temp < 0) {
+		SYS_LOG_DBG("Default poffset UR not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_POFFSET_DL_REG, APDS9960_DEFAULT_POFFSET_DL);
+	if (temp < 0) {
+		SYS_LOG_DBG("Default poffset DL not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG1_REG, APDS9960_DEFAULT_CONFIG1);
+	if (temp < 0) {
+		SYS_LOG_DBG("Default config1 not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_CONTROL_LDRIVE,
+		APDS9960_DEFAULT_LDRIVE);
+	if (temp < 0) {
+		SYS_LOG_DBG("LEDdrive not set");
+		return -EIO;
+	}
+
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_CONTROL_PGAIN,
+		(gain & APDS9960_DEFAULT_PGAIN));
+	if (temp < 0) {
+		SYS_LOG_DBG("proxy Gain is not set");
+		return temp;
+	}
+
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_CONTROL_AGAIN,
+		(gain & APDS9960_DEFAULT_AGAIN));
+	if (temp < 0) {
+		SYS_LOG_DBG("Ambient Gain is not set");
+		return temp;
+	}
+
+	temp = apds9960_setproxint_lowthresh(dev, APDS9960_DEFAULT_PILT);
+	if (temp < 0) {
+		SYS_LOG_DBG("prox low threshold not set");
+		return temp;
+	}
+
+	temp = apds9960_setproxint_highthresh(dev, APDS9960_DEFAULT_PIHT);
+	if (temp < 0) {
+		SYS_LOG_DBG("prox high threshold not set");
+		return temp;
+	}
+
+	temp = apds9960_setlightint_lowthresh(dev, APDS9960_DEFAULT_AILT);
+	if (temp < 0) {
+		SYS_LOG_DBG("light low threshold not set");
+		return temp;
+	}
+
+	temp = apds9960_setlightint_highthresh(dev, APDS9960_DEFAULT_AIHT);
+	if (temp < 0) {
+		SYS_LOG_DBG("light high threshold not set");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PERS_REG, APDS9960_DEFAULT_PERS);
+	if (temp < 0) {
+		SYS_LOG_DBG("ALS interrupt persistence not set ");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONFIG2_REG, APDS9960_DEFAULT_CONFIG2);
+	if (temp < 0) {
+		SYS_LOG_DBG("clear diode saturation interrupt is not enabled");
+		return temp;
+	}
+
+	temp = apds9960_proxy_setup(dev, gain);
+	if (temp < 0) {
+		SYS_LOG_DBG("Failed to setup proxymity functionality");
```

Comment:
For above feedback. I wrote interrupt bit separately because later I want to implement 
#if INTERRUPT_IS_ON
write interrupt bit 1
#else 
write interrupt bit 0

That's why I haven't combined with above transaction. All other sensor driver do the same.